tools/mo/openvino/tools/mo/back/ResultRename.py

```diff
@@ -13,18 +13,33 @@ class ResultRename(BackReplacementPattern):
     enabled = False
 
     def find_and_replace_pattern(self, graph: Graph):
+        names_set = set()
+        for node in graph.get_op_nodes():
+            if node.has_valid('name'):
+                names_set.add(node['name'])
+
         for node in graph.get_op_nodes(type='Result'):
             if node.in_ports():
                 prev_node_out_port = node.in_port(0).get_connection().get_source()
                 tensor_names = prev_node_out_port.get_tensor_names()
                 # Graph may contain Result nodes with names equal to input tensors and
                 # renaming in this case is not needed. The example of such situation is
                 # IR reader check when graph is read with correct Result names.
-                if tensor_names and node.soft_get('name') == tensor_names[0]:
+                if not tensor_names:
+                    continue
```

Comment:
In this case we also need to compose custom name for Result node using `/sink_port_` suffix as below

Reply:
Fixed.